components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -198,7 +199,26 @@ public AuthenticatorFlowStatus process(HttpServletRequest request,
             } finally {
                 AutoLoginUtilities.removeAutoLoginCookieInResponse(response, autoLoginCookie);
             }
+        } else if (runtimeParams.containsKey(USER_NAME) && runtimeParams.containsKey(PASSWORD)) {
+            processAuthenticationResponse(request, response, context);
+            if (!context.getSequenceConfig().getApplicationConfig().isSaaSApp()) {
+                String userDomain = context.getSubject().getTenantDomain();
+                String tenantDomain = context.getTenantDomain();
+                if (!StringUtils.equals(userDomain, tenantDomain)) {
+                    context.setProperty(FrameworkConstants.USER_TENANT_DOMAIN_MISMATCH, true);
+                    throw new AuthenticationFailedException(
+                            FrameworkErrorConstants.ErrorMessages.MISMATCHING_TENANT_DOMAIN.getCode(),
+                            FrameworkErrorConstants.ErrorMessages.MISMATCHING_TENANT_DOMAIN.getMessage(),
+                            context.getSubject());
+                }
+            }
+
+            request.setAttribute(FrameworkConstants.REQ_ATTR_HANDLED, true);
+            context.setProperty(FrameworkConstants.LAST_FAILED_AUTHENTICATOR, null);
+            publishAuthenticationStepAttempt(request, context, context.getSubject(), true);
+            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
```

Comment:
lets set skip prompt to true in the context and call the super.process method. it will take the code through the same flow and avoid duplication.

https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java#L106-L128

Reply:
Addressed the comment: 441980b23350cd06554e7aabe47bf7252765ccf1